scope.go

```diff
@@ -305,6 +376,9 @@ func (s *scope) Close() error {
 	if closer, ok := s.reporter.(io.Closer); ok {
 		return closer.Close()
 	}
+	if closer, ok := s.cachedReporter.(io.Closer); ok {
+		return closer.Close()
+	}
```

Comment:
If we were using the baseStatsReporter approach this could also just be:

```go
 	if closer, ok := s.baseReporter.(io.Closer); ok {
  		return closer.Close()		  		return closer.Close()
  	}
```